beta-pipelines/multiome/atac.wdl

```diff
@@ -0,0 +1,226 @@
+    Int? trim_disk_size
+    String adapter_seq_read1
+    String adapter_seq_read2
+
+    # BWA parameters
+    File tar_bwa_reference
+    String read_group_id = "RG1"
+    String read_group_sample_name = "RGSN1"
+    Int bwa_threads = 16
+    Int? bwa_disk_size
+    Int bwa_mem_size = 8
+
+    # Output prefix/base name for all intermediate files and pipeline outputs
+    String output_base_name
+
+    # script for monitoring tasks 
+    File monitoring_script
+  }
+
+  parameter_meta {
+    fastq_gzipped_input_read1: "read 1 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name seperated by colon"
+    fastq_gzipped_input_read2: "read 2 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name separated by colon"
+    min_length: "minimum length for trimming. Reads that are too short even before adapter removal are also discarded"
+    quality_cutoff: "cutadapt option to trim low-quality ends from reads before adapter removal"
+    adapter_seq_read1: "cutadapt option for the sequence adapter for read 1 fastq"
+    adapter_seq_read2: "cutadapt option for the sequence adapter for read 2 fastq"
+    tar_bwa_reference: "the pre built tar file containing the reference fasta and corresponding reference files for the BWA aligner"
+    read_group_id: "the read group id to be added upon alignment"
+    read_group_sample_name: "the read group sample to be added upon alignment"
+    output_base_name: "base name to be used for the pipelines output and intermediate files"
+    monitoring_script : "script to monitor resource comsumption of tasks"
+    bwa_threads: "the number of threads to use during bwa alignment"
+    bwa_disk_size : "disk size used in bwa alignment step"
+    bwa_mem_size : "the size of memory used during bwa alignment"
+    trim_disk_size : "disk size used in trimming adapters step"
+    trim_mem_size: "the size of memory used during trimming adapters step"
+  }
+
+  call TrimAdapters {
+    input:
+      fastq_input_read1 = fastq_gzipped_input_read1,
+      fastq_input_read2 = fastq_gzipped_input_read2,
+      min_length = min_length,
+      quality_cutoff = quality_cutoff,
+      adapter_seq_read1 = adapter_seq_read1,
+      adapter_seq_read2 = adapter_seq_read2,
+      output_base_name = output_base_name,
+      mem_size = trim_mem_size, 
+      disk_size = trim_disk_size, 
+      monitoring_script = monitoring_script
+  }
+
+  call BWAPairedEndAlignment {
+    input:
+      fastq_input_read1 = TrimAdapters.fastq_trimmed_adapter_output_read1,
+      fastq_input_read2 = TrimAdapters.fastq_trimmed_adapter_output_read2,
+      tar_bwa_reference = tar_bwa_reference,
+      read_group_id = read_group_id,
+      read_group_sample_name = read_group_sample_name,
+      nthreads = bwa_threads,
+      mem_size = bwa_mem_size,
+      disk_size = bwa_disk_size,
+      output_base_name = output_base_name,
+      monitoring_script = monitoring_script
+    }
```

Comment:
Are there outputs for the workflow at this stage?

Reply:
The sam file and the monitoring log are outputs of the bwa task. Should we include other outputs for this task?

Reply:
The SAM file from the example run is 50 GB. The SnapATAC2 tool requires BAM files. Should we just do the conversion to BAM in this task?

Reply:
I think we should go ahead and do it in the BWA task.